### ISSUE TITLE
DisplayName value as part of @Test?

### ISSUE THREAD

**Issue by brsanthu** [junit-team/junit5#614]
May be this was discussed earlier but I couldn't find any references. As we are building new api, is there a reason why we cannot specify test name as part of test itself? Like below

Instead of
```
@Test
@DisplayName("test something")
void test() throws Exception {
    //test
}

```
Can we write this as,
```
@Test("test something")
void test() throws Exception {
    //test
}
```

This would save one additional annotation we need to specify for each test.

**Comment by marcphilipp** [comment]
The reason it's a separate annotation is that is can also be used to specify display names for top-level or nested classes as well as `@TestFactory` methods.

**Comment by brsanthu** [comment]
Thank you for the comment. Then is it possible to overload that if there is a @DisplayName, that will take precedence over name within the @Test?

**Comment by sormuras** [comment]
I do like the optional `@Test("use this string if no @DisplayName is attached")` notation.

**Comment by bechte** [comment]
You could have your own

**Comment by bechte** [comment]
You could have your own

**Comment by bechte** [comment]
Sorry my iPhone crashed during posting and resulted in strange comments and closing this issue. I reopened it.

**Comment by bechte** [comment]
We did not want to have the @Test annotation spoiled with several concerns and also wanted to have @DisplayName to be supported on different level as mentioned previously by Marc. Bringing it back to life with a fallback or overriding mechanism could lead to misunderstanding in reading a test case which I think we should avoid.

How do the others feel?

**Comment by sbrannen** [comment]
I agree with @marcphilipp and @bechte regarding the rationale for the status quo.

Furthermore, I am strongly opposed to looking up anything in multiple ways with regard to annotations.

Unlike the Spring Framework -- which supports merged annotation attribute overrides within annotation hierarchies -- JUnit's annotation search algorithms only support simple look-ups of a given annotation which may potentially be meta-present (i.e., present as a meta-annotation). 

In other words, if JUnit Jupiter were to support look-ups of the display name via the `@Test` annotation, that would introduce a special case and a whole new can of worms that we want to avoid both for the sake of simplicity and maintenance.

**Comment by sbrannen** [comment]
> Thank you for the comment. Then is it possible to overload that if there is a `@DisplayName`, that will take precedence over name within the `@Test`?

This is exactly the type of _special case code_ that we want to avoid.

**Comment by sbrannen** [comment]
@bechte 

> You could have your own

I'm not quite sure what you meant by that.

Care to expound (taking into consideration the shortcomings of JUnit's annotation search algorithms that I've already mentioned)?

**Comment by marcphilipp** [comment]
I'm closing this issue since the team agrees to keep the status quo.

**Comment by brsanthu** [comment]
Thank you team for discussing this.

**Comment by nicolaiparlog** [comment]
Beating a dead horse here... but would this not be an incentive to create an extension point that names tests? Then @brsanthu could create their own `@Test` annotation that does indeed define the test's name. If this seems to be worth pursuing, I can open a new issue.

**Comment by marcphilipp** [comment]
Creating an extension point for naming tests sounds nice but it isn't straight-forward. All current extensions points act during execution. Naming tests happens during discovery, so we cannot provide an `TestExtensionContext`. Maybe I'm think to complicated here but what would we pass an extension to determine the name? I don't want to introduce dependencies to platform modules, so neither `TestIdentifier` nor `TestDescriptor` qualify. I'm open for suggestions, though. 🙂

**Comment by nicolaiparlog** [comment]
Sounds too complicated to pursue just on a whim. :wink: I won't open an issue then.

**Comment by sbrannen** [comment]
related to #1440
